src/components/MapView/index.tsx

```diff
@@ -58,7 +76,11 @@ function MapView({ classes }: MapViewProps) {
   const loading = layersLoading || datesLoading;
 
   useEffect(() => {
+    // initial load, need available dates and boundary layer
+    const boundaryLayer = getBoundaryLayerSingleton();
     dispatch(loadAvailableDates());
+    dispatch(addLayer(boundaryLayer));
+    dispatch(loadLayerData({ layer: boundaryLayer }));
```

Comment:
Yes I agree! It would be cleaner to put it in `BoundaryLayer`, I actually had it there initially however the two issues you mentioned are exactly what happened: 

1. The loading screen happens twice instead of just one single loading screen at the beginning.

2. If you are quick enough to add a layer via the nav bar during the first loading screen, it will not display after the loading finishes. You'll have to re-toggle the switch to display the layer. 

As a quick fix I just moved the loading into the map view so these were fixed, but it definitely isn't too logical to keep it that way.

I like the idea where we hard code it into the tree, kind of like how it was done last time. I'll try play around with it and see if I can prevent the aforementioned issues, otherwise I'll document the existing code with hopes in the future it can be resolved.

Reply:
@bencpeters After experimenting a little, I've concluded its quite hard to try keep the loading in the `BoundaryLayer` component.

The reason we get two loading screens (one after the other) is because Mapbox decides to render its children incredibly late - after most of the base map has finished rendering. Therefore, `BoundaryLayer` never gets the chance to dispatch a load early. 

This makes the program inefficient since it can't start loading `admin_boundaries` while everything else is loading, since it must wait for mapbox to finish. This also opens the opportunity for a user to try load another layer simultaneously with `BoundaryLayer`, causing an error and forcing the user to re-toggle the layer to view it. 

I think the best approach is to load the layer data within the `MapView` `useEffect`, but adding extra documentation to indicate why. I'll still also add logic to `BoundaryLayer` to load its own data just for convention sake and in case we decide to use multiple in the future (this will come with added logic to the `loadLayerData` thunk to prevent loading the same thing twice). 

If you have another suggestion, let me know 👍